src/main/java/org/openhab/tools/analysis/checkstyle/api/AbstractStaticCheck.java

```diff
@@ -92,6 +100,24 @@ protected int findLineNumber(FileText fileContent, String searchedText, int star
      * Finds the first occurrence of a text in a list of text lines representing the file content and
      * returns the line number, where the text was found
      *
+     *
+     * @param fileContent - represents the text content
+     * @param searchedText - the text that we are looking for
+     * @param startLineNumber - the line number from which the search starts exclusive, to start the
+     *            search of the beginning of the text the startLineNumber should be 0
+     * @return the number of the line starting from 1, where the searched text occurred for the first
+     *         time
+     * @throws NoResultException when no match was found
+     */
+    protected int findLineNumber(FileText fileContent, String searchedText, int startLineNumber)
+            throws NoResultException {
+        return findLineNumberFormatted(fileContent, searchedText, startLineNumber, null);
```

Comment:
I don't like `null` I would have prefered a no-op `LineFormatterCallback` if we go for this solution than you also do not need the null check, just let it explode if you are called with a `null`

Reply:
Thank you for this suggestion, it does seem to make the code more robust. I will rewrite it in the suggested way.